1-Draft/RFC00XX-Native-Command-Error-Handling.md

```diff
@@ -0,0 +1,334 @@
+
+- The value will default to `Continue` for compatibility with existing behavior.
+- For non-zero exit codes and except for the value `Ignore`, an `ErrorRecord` will be added to
+  `$Error` that wraps the exit code and the command executed that returned the exit code.
+- Initially, only the existing values of `$ErrorActionPreference` will be supported in
+  `$PSNativeCommandErrorAction` as described in the table below.
+
+### Valid values for `$PSNativeCommandErrorAction`
+
+| Value           | Definition
+----------------  | -------------------
+| Break           | Enter the debugger when an error occurs or when an exception is raised.
+| Continue        | Displays the error message and continues executing.
+| Ignore          | Suppresses the error message and continues to execute the command.
+| Inquire         | Displays the error message and asks you whether you want to continue.
+| SilentlyContinue| No effect. The error message isn't displayed and execution continues without interruption.
+| Stop            | Displays the error message and stops executing. In addition to the error generated, the Stop value generates an ActionPreferenceStopException object to the error stream. stream
+| Suspend         | Automatically suspends a workflow job to allow for further investigation.
+
+### Preference variable resolves error handling conflicts
+
+In cases where an existing script already handles non-zero native command errors, the preference
+variable `$PSNativeCommandErrorAction` may be set to `Continue`. Error handling behavior of the
+PowerShell cmdlets is handled separately with `$ErrorActionPreference`.
+
+### Error object
+
+The reported error record object will be the new type: `NativeCommandException` with the following details:
+
+| Property        | Definition
+----------------  | -------------------
+| ExitCode:      |  The exit code of the failed command.
+| ErrorID:       | `"Program {0} ended with non-zero exit code {1}"`, with the command name and the exit code, from resource string `ProgramFailedToComplete`.
+| ErrorCategory: | `ErrorCategory.NotSpecified`.
+| object:         | exit code
+| Source:        | The full path to the application
+| ProcessInfo    | details of failed command including path, exit code, and PID
+
+## Alternative Approaches and Considerations
+
+### Native commands should respect $ErrorActionPreference
+
+Native commands should respect $ErrorActionPreference and not need $PSNativeCommandErrorAction. This
+could be released as an experimental feature.
+
+### Extending $PSNativeCommandErrorAction
+
+For users that prefer to set a single preference variable that affects both PowerShell and native
+commands, The values of `$PSNativeCommandErrorAction` may be extended to include
+`MatchErrorActionPreference`, which should apply the `$ErrorActionPreference` setting to native
+commands.
+
+- A conversion will occur between `$PSNativeCommandErrorAction` and `$ErrorActionPreference` values,
```

Comment:
If `$PSNativeCommandErrorAction` is set to `MatchErrorActionPreference` then the value of `$ErrorActionPreference` will define the behavior for native command errors.